tests/end2end/features/private.feature

```diff
@@ -0,0 +1,112 @@
+        And I run :command-accept
+        # Then the error should not be shown again
+
+    ## https://github.com/qutebrowser/qutebrowser/issues/1219
+
+    @qtwebkit_ng_skip: private browsing is not implemented yet
+    Scenario: Sharing cookies with private browsing
+        When I open cookies/set?qute-test=42 without waiting in a private window
+        And I wait until cookies is loaded
+        And I open cookies in a new tab
+        And I set general -> private-browsing to false
+        Then the cookie qute-test should be set to 42
+
+    Scenario: Opening private window with :navigate increment
+        # Private window handled in commands.py
+        When I open data/numbers/1.txt in a private window
+        And I run :window-only
+        And I run :navigate -w increment
+        And I wait until data/numbers/2.txt is loaded
+        Then the session should look like:
+            windows:
+            - private: True
+              tabs:
+              - history:
+                - url: http://localhost:*/data/numbers/1.txt
+            - private: True
+              tabs:
+              - history:
+                - url: http://localhost:*/data/numbers/2.txt
+
+    Scenario: Opening private window with :navigate next
+        # Private window handled in navigate.py
+        When I open data/navigate in a private window
+        And I run :window-only
+        And I run :navigate -w next
+        And I wait until data/navigate/next.html is loaded
+        Then the session should look like:
+            windows:
+            - private: True
+              tabs:
+              - history:
+                - url: http://localhost:*/data/navigate
+            - private: True
+              tabs:
+              - history:
+                - url: http://localhost:*/data/navigate/next.html
+
+    Scenario: Opening private window with :tab-clone
+        When I open data/hello.txt in a private window
+        And I run :window-only
+        And I run :tab-clone -w
+        And I wait until data/hello.txt is loaded
+        Then the session should look like:
+            windows:
+            - private: True
+              tabs:
+              - history:
+                - url: http://localhost:*/data/hello.txt
+            - private: True
+              tabs:
+              - history:
+                - url: http://localhost:*/data/hello.txt
+
+    Scenario: Opening private window via :click-element
+        When I open data/click_element.html in a private window
+        And I run :window-only
+        And I run :click-element --target window id link
+        And I wait until data/hello.txt is loaded
+        Then the session should look like:
+            windows:
+            - private: True
+              tabs:
+              - history:
+                - url: http://localhost:*/data/click_element.html
+            - private: True
+              tabs:
+              - history:
+                - url: http://localhost:*/data/hello.txt
```

Comment:
You mean `:back` / `:forward`? That's entirely handled for the backend (qutebrowser just calls the back/forward methods on the web view), so I don't think it's useful to add a test for this.